nsfrb/searching.py

```diff
@@ -287,9 +584,304 @@ def snr_vs_RA_DEC(image_tesseract,boxcar,gridsize,plot=False,width=1,TMPCOORDS=[
+    
+        print("Starting boxcar filtering with " + str(nwidthtrials) + " trials...",file=fout)
+        #PSF parameters
+        maxs = []
+        maxs2 = []
+        for w in range(nwidthtrials):
+            for d in range(nDMtrials):
+                image_tesseract_binned[:,:,w,d] = snr_vs_RA_DEC_new(image_tesseract_dedisp[:,:,:,d],widthtrials[w],noiseth=noiseth,output_file=output_file) 
+                if d ==0 and plot:
+                    maxs.append(image_tesseract_binned[15, 16,w,d])
+                elif plot:
+                    maxs2.append(image_tesseract_binned[15, 16,w,d])
+        print("Done!",file=fout)    
+        print("---> " + str(np.sum(np.isnan(image_tesseract_binned))),file=fout)
+
+        if plot:
+            plt.figure(figsize=(12,6))
+            plt.plot(widthtrials,maxs,'o-')
+            #plt.plot(widthtrials,maxs2,'o-')
+            #plt.plot(np.arange(1,10),maxs[np.argmin(np.abs(widthtrials-5))]*np.sqrt(5/np.arange(1,10)),color='red')
+            #plt.plot(np.arange(1,10),maxs[np.argmin(np.abs(widthtrials-5))]*np.sqrt(np.arange(1,10)/5),color='blue')
+            plt.show()
+        
+
+
+        print("Searching for candidates with S/N > " + str(SNRthresh) + "...",file=fout)
+        #find candidates above SNR threshold
+        condition = (image_tesseract_binned>SNRthresh).flatten()
+        ncands = np.sum(condition)
+        canddec_idxs,candra_idxs,candwid_idxs,canddm_idxs=np.unravel_index(np.arange(gridsize_DEC*gridsize_RA*nDMtrials*nwidthtrials)[condition],(gridsize_DEC,gridsize_RA,nwidthtrials,nDMtrials))#[1].shape
+    
+        canddecs = DEC_axis[canddec_idxs]
+        candras = RA_axis[candra_idxs]
+        candwids = widthtrials[candwid_idxs]
+        canddms = DM_trials[canddm_idxs]
+        candsnrs = image_tesseract_binned.flatten()[condition]
+    
+        candidxs = [(raidx_offset + candra_idxs[i],decidx_offset + canddec_idxs[i],candwid_idxs[i],dm_offset + canddm_idxs[i],candsnrs[i]) for i in range(ncands)]
+        cands = [(candras[i],canddecs[i],candwids[i],canddms[i],candsnrs[i]) for i in range(ncands)]
+
+        #make a dictionary for easy plotting of results
+        canddict['ra_idxs'] = copy.deepcopy(candra_idxs + raidx_offset)
+        canddict['dec_idxs'] = copy.deepcopy(canddec_idxs + decidx_offset)
+        canddict['wid_idxs'] = copy.deepcopy(candwid_idxs)
+        canddict['dm_idxs'] = copy.deepcopy(canddm_idxs + dm_offset)
+        canddict['ras'] = copy.deepcopy(candras)
+        canddict['decs'] = copy.deepcopy(canddecs)
+        canddict['wids'] = copy.deepcopy(candwids)
+        canddict['dms'] = copy.deepcopy(canddms)
+        canddict['snrs'] = copy.deepcopy(candsnrs)
+
+    print("Done! Found " + str(ncands) + " candidates",file=fout)
+    if output_file != "":
+        fout.close()
+    return candidxs,cands,image_tesseract_binned,image_tesseract_filtered,canddict,DM_trials
+
+
+
+### *** DEPRECATED 3/20/2024
```

Comment:
@mbsherma Need to remove dead code here

Reply:
Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)